### PR TITLE
Small bits

### PR DESCRIPTION
**Pull request by vext01**
Here are a handful of changes that improve the overall code quality of hwtracer.

Some found by clippy, some by me.

See individual commit messages.

### CODE REVIEW

File: src/backends/perf_pt/mod.rs [hunk 1]
```diff
@@ -60,8 +60,11 @@ extern "C" {
 #[repr(C)]
 #[derive(Debug)]
 pub struct PerfPTTrace {
+    // The trace buffer.
     buf: *mut u8,
+    // The length of the trace (in bytes).
     len: u64,
+    // `buf`'s allocation size (in bytes), <= `len`.
```
Comment (ltratt) [inline src/backends/perf_pt/mod.rs]: I think this should be `>=`?

**Comment by ltratt** [comment]
OK, I think we can probably squash a bit before merging?

**Comment by vext01** [comment]
I'd actually like to keep them separate, as they are all unrelated. Is that OK?

**Comment by ltratt** [comment]
OK.

**Comment by vext01** [comment]
Merge away :)

**Pull request merged**